lemur/endpoints/models.py

```diff
@@ -150,7 +150,7 @@ def sni_certificates(self):
     @sni_certificates.setter
     def sni_certificates(self, certs):
         """Sets the SNI certificates associated with the endpoint."""
-        self.certificates_assoc = [assoc for assoc in self.certificates_assoc if not assoc.primary]
+        self.certificates_assoc = [assoc for assoc in self.certificates_assoc if assoc.primary]
```

Comment:
I think Im having trouble wrapping my head around this logic. For this method I would think we are trying to get all the the non-primary certificates. But now the logic reads like we are only interested in the primary cert.

Reply:
This method is a "setter" which means it's used for replacing all of the sni certificates (and only the sni certificates) attached to the endpoint. The intent of this line is to ensure the SNI certificates can be set without removing the primary certificate.

```
endpoint.sni_certificates = [ ... ] # should not remove primary certificate
```

Reply:
I see, so this line is setting `self.certificates_assoc` to only be the primary cert and then the subsequent loop is setting all the SNI certs.